spec/models/miq_queue_spec.rb

```diff
@@ -986,4 +986,97 @@ def queue_items
       )
     end
   end
+
+  describe ".messaging_client_options" do
+    before do
+      config_file = Rails.root.join("config", "messaging.yml")
+      FileUtils.cp(Rails.root.join("config", "messaging.kafka.yml"), config_file) unless config_file.file?
+    end
+
+    context "with ENV" do
+      around do |example|
+        old_env = ENV
+        ENV["MESSAGING_HOSTNAME"] = "server.example.com"
+        ENV["MESSAGING_PASSWORD"] = "password"
+        ENV["MESSAGING_PORT"]     = "9092"
+        ENV["MESSAGING_USERNAME"] = "admin"
+
+        example.run
+
+        ENV.replace(old_env)
+      end
+
+      context "prefers settings from ENV when they exist" do
+        it "with clear text password" do
+          expect(MiqQueue.send(:messaging_client_options)).to eq(
+            :encoding => "json",
+            :host     => "server.example.com",
+            :password => "password",
+            :port     => 9092,
+            :protocol => nil,
+            :username => "admin"
+          )
+        end
+
+        it "with encrypted password" do
+          ENV["MESSAGING_PASSWORD"] = MiqPassword.encrypt(ENV["MESSAGING_PASSWORD"])
+
+          expect(ENV["MESSAGING_PASSWORD"]).not_to eq("password")
+          expect(MiqQueue.send(:messaging_client_options)).to eq(
+            :encoding => "json",
+            :host     => "server.example.com",
+            :password => "password",
+            :port     => 9092,
+            :protocol => nil,
+            :username => "admin"
+          )
+        end
+      end
+
+      it "prefers settings from file if any ENV vars are missing" do
+        ENV.delete("MESSAGING_PORT")
+
+        expect(MiqQueue.send(:messaging_client_options)).to eq(
+          :encoding => "json",
+          :host     => "localhost",
+          :password => "smartvm",
+          :port     => 9092,
+          :protocol => nil,
+          :username => "admin"
+        )
+      end
+    end
+
+    context "prefers settings from file when ENV vars are missing" do
+      it "with clear text password" do
+        expect(MiqQueue.send(:messaging_client_options)).to eq(
+          :encoding => "json",
+          :host     => "localhost",
+          :password => "smartvm",
+          :port     => 9092,
+          :protocol => nil,
+          :username => "admin"
+        )
+      end
+
+      it "with encrypted password" do
+        allow(YAML).to receive(:load_file).and_call_original
+        expect(YAML).to receive(:load_file).with(MiqQueue::MESSAGING_CONFIG_FILE).and_wrap_original do |m, *args|
+          m.call(*args).tap do |i|
+            i["test"]["password"] = MiqPassword.try_encrypt(i["test"]["password"])
+            expect(i["test"]["password"]).to start_with("v2:{")
```

Comment:
There should be a `be_encrypted` helper method for this

`expect(i["test"]["password"]).to be_encrypted`

Reply:
Oh yeah, forgot about that